src/buffer/ring_buffer.rs

```diff
@@ -240,6 +255,12 @@ mod if_alloc {
             assert!(self.buffer.len() > 0);
             self.buffer.pop_front().unwrap()
         }
+
+        #[inline]
+        fn peek(&self) -> &Self::Item {
+            assert!(self.buffer.len() > 0);
```

Comment:
I think this is not necessary given that `unwrap` is used in the next line. If the intention is to add more context information `unwrap` could be replaced with `.expect`